test/TestApps/ToastNotificationsTestApp/main.cpp

```diff
@@ -25,23 +25,115 @@ bool UnregisterBackgroundActivationTest()
     return true;
 }
 
-winrt::ToastNotification GetToastNotification()
+winrt::ToastNotification CreateToastNotification(winrt::hstring message)
 {
-    winrt::hstring xmlPayload{ L"<toast>intrepidToast</toast>" };
+    winrt::hstring xmlPayload{ L"<toast>" + message + L"</toast>" };
 
     winrt::XmlDocument xmlDocument{};
     xmlDocument.LoadXml(xmlPayload);
 
     return winrt::ToastNotification(xmlDocument);
 }
 
+winrt::ToastNotification CreateToastNotification()
+{
+    return CreateToastNotification(L"intrepidToast");
+}
+
+void EnsureNoActiveToasts()
+{
+    auto removeAllAsync = winrt::ToastNotificationManager::Default().RemoveAllAsync();
+    if (removeAllAsync.wait_for(std::chrono::seconds(300)) != winrt::Windows::Foundation::AsyncStatus::Completed)
+    {
+        removeAllAsync.Cancel();
+        THROW_HR_MSG(E_FAIL, "Failed to remove all active toasts");
+    }
+}
+
+bool VerifyToastIsActive(UINT32 expectedToastId)
+{
+    auto retrieveNotificationsAsync{ winrt::ToastNotificationManager::Default().GetAllAsync() };
+    if (retrieveNotificationsAsync.wait_for(std::chrono::seconds(300)) != winrt::Windows::Foundation::AsyncStatus::Completed)
+    {
+        retrieveNotificationsAsync.Cancel();
+        return false;
+    }
+
+    auto notifications = retrieveNotificationsAsync.get();
+
+    bool found{ false };
+    for (auto notification : notifications)
+    {
+        if (notification.ToastId() == expectedToastId)
+        {
+            found = true;
+            break;
+        }
+    }
+
+    return found;
+}
+
+bool VerifyToastNotificationIsValid(const winrt::ToastNotification& expected, const winrt::ToastNotification& actual)
+{
+    if (expected.Tag() != actual.Tag())
+    {
+        return false;
+    }
+
+    if (expected.Group() != actual.Group())
+    {
+        return false;
+    }
+
+    if (expected.ToastId() != actual.ToastId())
+    {
+        return false;
+    }
+
+    auto expectedPayload = expected.Payload().GetElementsByTagName(L"toast").GetAt(0).GetXml();
+    auto actualPayload = actual.Payload().GetElementsByTagName(L"toast").GetAt(0).GetXml();
+    if (expectedPayload != actualPayload)
+    {
+        return false;
+    }
+#if 0
+    // ELx - TODO implement this progress data support is available.
+    if (VerifyProgressData(expected.ProgressData(), actual.ProgressData()))
+    {
+        return false;
+    }
+#endif
+    if (expected.ExpirationTime() != actual.ExpirationTime())
+    {
+        return false;
+    }
+
+    if (expected.ExpiresOnReboot() != actual.ExpiresOnReboot())
+    {
+        return false;
+    }
+
+    if (expected.Priority() != actual.Priority())
+    {
+        return false;
+    }
+
+    if (expected.SuppressDisplay() != actual.SuppressDisplay())
+    {
+        return false;
+    }
```

Comment:
We don't have to keep these checks since they are transient properties. Maybe just add a comment saying why we aren't verifying them.

Reply:
I'll adjust them to check for default values.